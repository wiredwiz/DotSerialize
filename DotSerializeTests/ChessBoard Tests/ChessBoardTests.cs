using System;
using ApprovalTests;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using Org.Edgerunner.DotSerialize.Tests.DataTypes;

namespace Org.Edgerunner.DotSerialize.Tests.Chessboard_Unit_Tests
{
   [TestClass]
   public class ChessBoardTests
   {
      private const string IntialBoardSerializesProperly_Approved =
         "ChessBoardTests.InitialBoardSerializesProperly.approved.xml";

      [TestCleanup]
      private void CleanUp()
      {
         Utilities.DeleteFile(IntialBoardSerializesProperly_Approved);
         //Utilities.DeleteFile(SerializeDogOmittingTypesResultsInProperOutput_Approved);
         //Utilities.DeleteFile(SerializeCatOmittingReferentialIntegrityResultsInProperOutput_Approved);
         //Utilities.DeleteFile(SerializeCatDECultureResultsInProperOutput_Approved);
         //Utilities.DeleteFile(SerializeCatWithoutMapResultsInProperOutput_Approved);
         //Utilities.DeleteFile(SerializeCatWithMap1ResultsInProperOutput_Approved);
         //Utilities.DeleteFile(SerializeCatWithMap2ResultsInProperOutput_Approved);
      }

      [TestInitialize]
      private void Setup()
      {
         Utilities.ExtractEmbeddedFile(IntialBoardSerializesProperly_Approved);
         //Utilities.ExtractEmbeddedFile(SerializeDogOmittingTypesResultsInProperOutput_Approved);
         //Utilities.ExtractEmbeddedFile(SerializeCatOmittingReferentialIntegrityResultsInProperOutput_Approved);
         //Utilities.ExtractEmbeddedFile(SerializeCatDECultureResultsInProperOutput_Approved);
         //Utilities.ExtractEmbeddedFile(SerializeCatWithoutMapResultsInProperOutput_Approved);
         //Utilities.ExtractEmbeddedFile(SerializeCatWithMap1ResultsInProperOutput_Approved);
         //Utilities.ExtractEmbeddedFile(SerializeCatWithMap2ResultsInProperOutput_Approved);
      }

      [TestMethod]
      public void InitialBoardSerializesProperly()
      {
         var board = new ChessBoard();
         var serializer = new Serializer();
         string xml = serializer.SerializeObject(board);
         Approvals.VerifyXml(xml);
      }

      [TestMethod]
      public void InitialBoardDeserializesProperly()
      {
         var board = new ChessBoard();
         var serializer = new Serializer() { Settings = new Settings { UseTypeConstructors = true } };
         string xml = serializer.SerializeObject(board);
         var board2 = serializer.DeserializeObject<ChessBoard>(xml);
         for (int i = 0; i < 8; i++)
            for (int j = 0; j < 8; j++)
               Assert.AreEqual(board.Position[i, j], board2.Position[i, j]);
      }
   }
}
