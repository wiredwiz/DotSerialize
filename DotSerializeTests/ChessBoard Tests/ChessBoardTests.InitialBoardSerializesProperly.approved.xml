<ChessBoard xmlns:dts="http://www.dotserialize.com/Schema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" dts:type="Org.Edgerunner.DotSerialize.Tests.DataTypes.ChessBoard, DotSerializeTests" dts:id="1" dts:src="true">
  <_Position dts:type="Org.Edgerunner.DotSerialize.Tests.DataTypes.ChessPiece[,], DotSerializeTests" dts:id="2" dts:src="true" dts:dim="8,8">
    <item dts:index="0,0" dts:type="Org.Edgerunner.DotSerialize.Tests.DataTypes.ChessPiece, DotSerializeTests" dts:id="3" dts:src="true">
      <_Type>Rook</_Type>
    </item>
    <item dts:index="0,1" dts:type="Org.Edgerunner.DotSerialize.Tests.DataTypes.ChessPiece, DotSerializeTests" dts:id="4" dts:src="true">
      <_Type>Knight</_Type>
    </item>
    <item dts:index="0,2" dts:type="Org.Edgerunner.DotSerialize.Tests.DataTypes.ChessPiece, DotSerializeTests" dts:id="5" dts:src="true">
      <_Type>Bishop</_Type>
    </item>
    <item dts:index="0,3" dts:type="Org.Edgerunner.DotSerialize.Tests.DataTypes.ChessPiece, DotSerializeTests" dts:id="6" dts:src="true">
      <_Type>Queen</_Type>
    </item>
    <item dts:index="0,4" dts:type="Org.Edgerunner.DotSerialize.Tests.DataTypes.ChessPiece, DotSerializeTests" dts:id="7" dts:src="true">
      <_Type>King</_Type>
    </item>
    <item dts:index="0,5" dts:type="Org.Edgerunner.DotSerialize.Tests.DataTypes.ChessPiece, DotSerializeTests" dts:id="8" dts:src="true">
      <_Type>Bishop</_Type>
    </item>
    <item dts:index="0,6" dts:type="Org.Edgerunner.DotSerialize.Tests.DataTypes.ChessPiece, DotSerializeTests" dts:id="9" dts:src="true">
      <_Type>Knight</_Type>
    </item>
    <item dts:index="0,7" dts:type="Org.Edgerunner.DotSerialize.Tests.DataTypes.ChessPiece, DotSerializeTests" dts:id="10" dts:src="true">
      <_Type>Rook</_Type>
    </item>
    <item dts:index="1,0" dts:type="Org.Edgerunner.DotSerialize.Tests.DataTypes.ChessPiece, DotSerializeTests" dts:id="11" dts:src="true">
      <_Type>Pawn</_Type>
    </item>
    <item dts:index="1,1" dts:type="Org.Edgerunner.DotSerialize.Tests.DataTypes.ChessPiece, DotSerializeTests" dts:id="12" dts:src="true">
      <_Type>Pawn</_Type>
    </item>
    <item dts:index="1,2" dts:type="Org.Edgerunner.DotSerialize.Tests.DataTypes.ChessPiece, DotSerializeTests" dts:id="13" dts:src="true">
      <_Type>Pawn</_Type>
    </item>
    <item dts:index="1,3" dts:type="Org.Edgerunner.DotSerialize.Tests.DataTypes.ChessPiece, DotSerializeTests" dts:id="14" dts:src="true">
      <_Type>Pawn</_Type>
    </item>
    <item dts:index="1,4" dts:type="Org.Edgerunner.DotSerialize.Tests.DataTypes.ChessPiece, DotSerializeTests" dts:id="15" dts:src="true">
      <_Type>Pawn</_Type>
    </item>
    <item dts:index="1,5" dts:type="Org.Edgerunner.DotSerialize.Tests.DataTypes.ChessPiece, DotSerializeTests" dts:id="16" dts:src="true">
      <_Type>Pawn</_Type>
    </item>
    <item dts:index="1,6" dts:type="Org.Edgerunner.DotSerialize.Tests.DataTypes.ChessPiece, DotSerializeTests" dts:id="17" dts:src="true">
      <_Type>Pawn</_Type>
    </item>
    <item dts:index="1,7" dts:type="Org.Edgerunner.DotSerialize.Tests.DataTypes.ChessPiece, DotSerializeTests" dts:id="18" dts:src="true">
      <_Type>Pawn</_Type>
    </item>
    <item dts:index="6,0" dts:type="Org.Edgerunner.DotSerialize.Tests.DataTypes.ChessPiece, DotSerializeTests" dts:id="19" dts:src="true">
      <_Type>Pawn</_Type>
    </item>
    <item dts:index="6,1" dts:type="Org.Edgerunner.DotSerialize.Tests.DataTypes.ChessPiece, DotSerializeTests" dts:id="20" dts:src="true">
      <_Type>Pawn</_Type>
    </item>
    <item dts:index="6,2" dts:type="Org.Edgerunner.DotSerialize.Tests.DataTypes.ChessPiece, DotSerializeTests" dts:id="21" dts:src="true">
      <_Type>Pawn</_Type>
    </item>
    <item dts:index="6,3" dts:type="Org.Edgerunner.DotSerialize.Tests.DataTypes.ChessPiece, DotSerializeTests" dts:id="22" dts:src="true">
      <_Type>Pawn</_Type>
    </item>
    <item dts:index="6,4" dts:type="Org.Edgerunner.DotSerialize.Tests.DataTypes.ChessPiece, DotSerializeTests" dts:id="23" dts:src="true">
      <_Type>Pawn</_Type>
    </item>
    <item dts:index="6,5" dts:type="Org.Edgerunner.DotSerialize.Tests.DataTypes.ChessPiece, DotSerializeTests" dts:id="24" dts:src="true">
      <_Type>Pawn</_Type>
    </item>
    <item dts:index="6,6" dts:type="Org.Edgerunner.DotSerialize.Tests.DataTypes.ChessPiece, DotSerializeTests" dts:id="25" dts:src="true">
      <_Type>Pawn</_Type>
    </item>
    <item dts:index="6,7" dts:type="Org.Edgerunner.DotSerialize.Tests.DataTypes.ChessPiece, DotSerializeTests" dts:id="26" dts:src="true">
      <_Type>Pawn</_Type>
    </item>
    <item dts:index="7,0" dts:type="Org.Edgerunner.DotSerialize.Tests.DataTypes.ChessPiece, DotSerializeTests" dts:id="27" dts:src="true">
      <_Type>Rook</_Type>
    </item>
    <item dts:index="7,1" dts:type="Org.Edgerunner.DotSerialize.Tests.DataTypes.ChessPiece, DotSerializeTests" dts:id="28" dts:src="true">
      <_Type>Knight</_Type>
    </item>
    <item dts:index="7,2" dts:type="Org.Edgerunner.DotSerialize.Tests.DataTypes.ChessPiece, DotSerializeTests" dts:id="29" dts:src="true">
      <_Type>Bishop</_Type>
    </item>
    <item dts:index="7,3" dts:type="Org.Edgerunner.DotSerialize.Tests.DataTypes.ChessPiece, DotSerializeTests" dts:id="30" dts:src="true">
      <_Type>Queen</_Type>
    </item>
    <item dts:index="7,4" dts:type="Org.Edgerunner.DotSerialize.Tests.DataTypes.ChessPiece, DotSerializeTests" dts:id="31" dts:src="true">
      <_Type>King</_Type>
    </item>
    <item dts:index="7,5" dts:type="Org.Edgerunner.DotSerialize.Tests.DataTypes.ChessPiece, DotSerializeTests" dts:id="32" dts:src="true">
      <_Type>Bishop</_Type>
    </item>
    <item dts:index="7,6" dts:type="Org.Edgerunner.DotSerialize.Tests.DataTypes.ChessPiece, DotSerializeTests" dts:id="33" dts:src="true">
      <_Type>Knight</_Type>
    </item>
    <item dts:index="7,7" dts:type="Org.Edgerunner.DotSerialize.Tests.DataTypes.ChessPiece, DotSerializeTests" dts:id="34" dts:src="true">
      <_Type>Rook</_Type>
    </item>
  </_Position>
</ChessBoard>